src/async_impl/upgrade.rs

```diff
@@ -0,0 +1,73 @@
+use std::pin::Pin;
+use std::task::{self, Poll};
+use std::{fmt, io};
+
+use futures_util::TryFutureExt;
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+/// An upgraded HTTP connection.
+pub struct Upgraded {
```

Comment:
I originally chose a separate type because I wasn't sure how much reqwest should expose hyper directly. Also to leave some space for possibly doing something different in reqwest.

Reply:
I don't mind changing it to just re-export hyper either

Reply:
@seanmonstar gently ping. let me know if you'd just like the hyper types re-exported.